### DISCUSSION TITLE
native increment/counter

### DISCUSSION THREAD

**Discussion by zidkim** [supabase/supabase#7635] [Feature Requests]
Hi Team,

As someone with a NoSQL background and little SQL knowledge/experience, it would be great to have native support/functionality for common operations such as incrementing a counter. I understand that there is a workaround using RPC, however from my perspective, I see it as an abstract workaround. Of course, I could get the current count by making a separate request, but that also seems quite bloated for such an operation.

I should note that the current process of adding a custom function is not difficult and the SQL language itself makes it relatively simple to understand. That being said, I mainly write this from my initial impressions and thought why can't something like this be supported natively.

Something like this would be great:

```
await supabase.from("team").update({
     points: supabase.helpers.increment
})
```

**Comment by silentworks** [comment]
Currently as you stated you can do this with an RPC however I have seen that some work has started on postgrest (which is what the supabase-js library uses under the hood for querying the database) to add this feature in. I can't remember exactly where the conversation is about this but @steve-chavez should be able to point you to it.

**Reply by zidkim** [reply]
I see I see, thanks for the info! I edited my question as you were writing your answer to note that the entire process itself is not difficult given the helpful threads, so not a big deal.

**Reply by silentworks** [reply]
I found the link to the thread in which my colleague @steve-chavez answered and spoke about what's currently being worked on. https://github.com/supabase/supabase/discussions/7613

**Reply by zidkim** [reply]
Thank you!
